src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -116,4 +117,77 @@ private User findUserToActivateOrThrowException(UUID userId,
+            + "        .container { width: 80%; margin: 0 auto; }\n"
+            + "        .header { background-color: #f4f4f4; padding: 20px; text-align: center; }\n"
+            + "        .content { padding: 20px; }\n"
+            + "        .footer { background-color: #f4f4f4; padding: 10px; text-align: center; font-size: 12px; }\n"
+            + "        .button {\n"
+            + "            display: inline-block;\n"
+            + "            padding: 10px 20px;\n"
+            + "            font-size: 16px;\n"
+            + "            color: #fff;\n"
+            + "            background-color: #1ae866;\n"
+            + "            text-decoration: none;\n"
+            + "            border-radius: 5px;\n"
+            + "            text-align: center;\n"
+            + "        }\n"
+            + "        .button:hover {\n"
+            + "            background-color: #15d67c;\n"
+            + "        }\n"
+            + "    </style>\n"
+            + "</head>\n"
+            + "<body>\n"
+            + "    <div class=\"container\">\n"
+            + "        <div class=\"header\">\n"
+            + "            <h1>Password Recovery</h1>\n"
+            + "        </div>\n"
+            + "        <div class=\"content\">\n"
+            + "            <p>Hello " + user.getUsername() + ",</p>\n"
+            + "            <p>We received a request to reset your password. Click the button below to create a new password:</p>\n"
+            + "            <p style=\"display: flex; flex: content; justify-content: center;\"><a href=\" " + recoverPasswordURL + "\" class=\"button\">Reset Password</a></p>\n"
+            + "            <p>If you did not request this change, please ignore this email.</p>\n"
+            + "            <p>Best regards,<br>The Support Team</p>\n"
+            + "        </div>\n"
+            + "        <div class=\"footer\">\n"
+            + "            <p>Megastore, Villa Maria, Cordoba, Argentina</p>\n"
+            + "        </div>\n"
+            + "    </div>\n"
+            + "</body>\n"
+            + "</html>";
+    emailService.sendEmail(email, "Recover Password", emailContent);
+  }
+
+  private User findUserByEmailOrThrowException(String email) {
+    return userRepository.findByEmailAndDeletedIsFalse(email).orElseThrow(() -> new RuntimeException("User not found"));
+  }
+
+  @Override
+  public void recoverPassword(UUID userId, String newPassword, UUID recoverPasswordToken) {
+    User user = findUserToRecoverPasswordOrThrowException(userId, recoverPasswordToken);
+    user.setPassword(passwordEncoder.encode(newPassword));
+    userRepository.save(user);
+  }
+
+  private User findUserToRecoverPasswordOrThrowException(UUID userId, UUID recoverPasswordToken) {
+    return userRepository.findByUserIdAndRecoverPasswordTokenIs(userId, recoverPasswordToken)
+            .orElseThrow(() -> new RuntimeException("Token is invalid or user not exists."));
```

Comment:
Acá deberíamos usar otro tipo de exception, un RuntimeException es MUY genérico, Spring tiene un tipo de exception que nos permite, además, asignarle un HTTP status al error

```suggestion
            .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "Token is invalid or user not exists."));
```